src/core/js/scriptLoader.js

```diff
@@ -8,6 +8,23 @@
         return false;
     })();
 
+    // Inject loading spinner into wrapper
+    domReady(function() {
+        var spinner = document.createElement("div");
+        var attr = document.createAttribute("class");
+        attr.value = "loading";
+        spinner.attributes.setNamedItem(attr);
+        spinner.innerHTML = 
+            '<style>' +
+            '.loading {position: fixed!important;background-color:transparent;text-align: center;}'+
+            '.loading .loader-gif {background-image: url(adapt/css/assets/ajax-loader.gif);background-repeat: no-repeat;background-position: center;width: 128px;height: 128px;position: absolute;top: 50%;left: 50%;margin-top: -64px;margin-left: -64px;}' +
+            '</style>' +
```

Comment:
yes that is correct. I added it as an inline style, so that browsers can render it before the main.css is loaded. Looks like Chrome waits until all css files are loaded.